opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -503,9 +500,130 @@ message DoubleHistogramDataPoint {
   // a boolean value which decides what type of intervals to use.
   repeated double explicit_bounds = 7;
 
+  // Multiple ways to define the bound sequence is supported. While explicit_bounds is one of the ways,
+  // for backward compatibility reason, it is not included in "oneof bound_type".
+  // Only one of explicit_bounds or bound_type should be present. When both are present, bound_type is ignored.
+  oneof bound_type {
+    LinearBounds linear_bounds = 9;
+    ExponentialBounds exponential_bounds = 10;
+    LogLinearBounds log_linear_bounds = 11;
+  }
   // (Optional) List of exemplars collected from
   // measurements that were used to form the data point
   repeated DoubleExemplar exemplars = 8;
+
+  // LinearBounds. Bounds can be generated via
+  // for (int i = 0; i < num_of_bounds; i++)
+  //     boundList.add(offset + width * i);
+  message LinearBounds {
+    double offset = 1;
+    double width = 2;
+    uint32 num_of_bounds = 3;
+  }
+
+  // ExponentialBounds. Bounds are on log scale.
+  // The bound sequence is stitched together using negative bounds, zero bound, and positive bounds.
+  // The general rules is: bound = reference * power(base, exponent)
+  // Reference and base are constants. Exponent is integer index of histogram buckets.
+  // Reference defaults to 1. Base should be greater than 1.
+  // Negative index is allowed. It results in a bound between 0 and the reference.
+  // Two sections are specified, one for positive numbers, one for negative numbers.
+  // A special field is used to specify if a bound of zero exists.
+  // The formula to generate explicit bounds here are for demonstration purpose only.
+  // If the message receiver natively stores exponential histograms, it will not need to generate explicit bounds.
+  // It will only need to record the exponential bound parameters.
+  message ExponentialBounds {
+    double reference = 1; // If not present, 1 is assumed.
+    double base = 2;
+
+    // The bound sequence starts with negative values generated via:
+    // for (int i = num_of_bounds_for_negative_numbers - 1; i >= 0; i--)
+    //    int exponent = i + index_offset_for_negative_numbers;
+    //    boundList.add( -1 * reference * power(base, exponent));
+    //
+    // Example of reference=1, base=2, index_offset_for_negative_numbers=-2, num_of_bounds_for_negative_numbers=6
+    // bounds:    -8 -4 -2 -1 -.5 -.25
+    // exponent:   3  2  1  0 -1  -2
+    // Note that the loop on i goes down, so that negative numbers with larger absolute values
+    // (but smaller arithmetic values) are added first.
+
+    sint32 index_offset_for_negative_numbers = 3;
```

Comment:
We've included several fields in the ExponentialBounds message to ensure that the bucket sizing can be asymmetical. (for example, so we can have more granularity for positive numbers than for negative numbers). Like this:
```
                                                   0
|________________________|________________|________|_|__|____|________|________________|
```

Is there a compelling reason to support asymmetric bounds?
It would simplify the message if we simply made the exponential bounds symmetric around their central point.

Reply:
see previous reply